app/scripts/migrations/054.js

```diff
@@ -0,0 +1,58 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Migrates preference tokens with 0 decimals typed as 'string' to 'number'
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const newState = state;
+
+  if (!newState.PreferencesController) {
+    return newState;
+  }
+
+  const { tokens } = newState.PreferencesController || [];
+  if (Array.isArray(tokens)) {
+    for (const token of tokens) {
+      if (token.decimals === '0') {
```

Comment:
Are you sure that `0` was the only possible string? :thinking:  We could check the type here instead.

Reply:
@Gudahtt - `0` was the only default decimal value I had seen occur during testing of numerous NFT's, but I think you're right that things should extend to check for strings in general. I can make needed updates here

Reply:
per our conversation, we are now doing general string primitive checks, as well as looking for an potential corruption that could result in being unable to perform a numerical conversion.